git-hub

```diff
@@ -816,61 +813,113 @@ class CloneCmd (object):
+            # the user to use those instead
+            for a in args.unknown_args:
+                if a in ('-o', '--origin') or a.startswith('-o', '--origin='):
+                    die("Please use --forkremote or --upstreamremote to name your "
+                        "remotes instead of using Git's `{}` option!", a)
+            # If we just forked the repo, GitHub might still be doing the actual
+            # fork, so cloning could fail temporarily. See
+            # https://github.com/sociomantic-tsunami/git-hub/issues/214
+            cls.git_retry_if(not args.triangular and forked,
+                    'clone',
+                    args.unknown_args + ['--origin', remote, '--', url, dest],
+                    'Cloning {} to {}'.format(url, dest))
+            # Complete the repository setup
+            os.chdir(dest)
+        else:
+            # We are inside working directory of a pre-cloned repository, just
+            # setup a remote and fetch it
+            added=cls.git_add_remote(remote, url)
+            if added:
+                cls.git_retry_if(not args.triangular and forked,
+                        'fetch', ['--', remote],
+                        'Fetching from {} ({})'.format(remote, url))
+
         if triangular:
+            fetchremote = args.forkremote
             remote_url = repo[urltype]
             git_config('remote.pushdefault', prefix='', value=fetchremote)
-        git_config('upstreamremote', value=args.upstreamremote)
-        git_config('forkremote', value=args.forkremote)
+        else:
+            fetchremote = args.upstreamremote
+            remote_url = repo['parent'][urltype] if not personal else repo[urltype]
+        validate_url(remote_url, urltype)
+        added=cls.git_add_remote(fetchremote, remote_url)
+        if added:
+            # We also need to retry in here, although is less likely since we
+            # already spent some time doing the previous clone
+            cls.git_retry_if(args.triangular and forked,
+                    'fetch', ['--', fetchremote],
+                    'Fetching from {} ({})'.format(fetchremote, remote_url))
         git_config('urltype', value=urltype)
+        git_config('forkremote', value=args.forkremote)
+        git_config('upstreamremote', value=args.upstreamremote)
         git_config('upstream', value=upstream)
-        validate_url(remote_url, urltype)
-        git('remote', 'add', '--', fetchremote, remote_url)
-        # We also need to retry in here, although is less likely since we
-        # already spent some time doing the previous clone
-        cls.git_retry_if(args.triangular and forked,
-                'fetch', ['--', fetchremote],
-                'Fetching from {} ({})'.format(fetchremote, remote_url))
-        run_hookscript('postclone', env=dict(
+        git_config('triangular', value='true' if triangular else 'false')
+        run_hookscript('post-setup-fork', env=dict(
```

Comment:
This makes a lot of sense, but is a bit unfortunate that this PR comes right after v2.1.0 was released, because it now constitute a breaking change :( (and we are quite strict about following semver).

Making it a non-breaking change should be easy enough though. We can just deprecate postclone and accept both until the next major is released and we can get rid of the old name. So if you can accept postclone as a hook name too and emit a warning saying the name is deprecated and will be removed in the future, everything should be fine to put this change in v2.2.0.

You can remove the old name from the documentation, don't worry about that.